### PR TITLE
Feature/assessment type endpoint

### PR DESCRIPTION
**Pull request by esotrope**
1. removes `administrationCondition`, `assessmentType` & `completeness` from `/reportOptions`
(kept underlying services & repos around for now)
2. adds `/assessmentTypes` endpoint to get user-specific assessment types

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/ReportOptions.java [hunk 1]
```diff
@@ -6,18 +6,11 @@
...
 public class ReportOptions {
 
-    private Set<CodedEntity> assessmentTypes;
     private Set<CodedEntity> subjects;
     private Set<CodedEntity> grades;
     private Set<Integer> schoolYears;
     private Set<CodedEntity> genders;
     private Set<CodedEntity> ethnicities;
-    private Set<CodedEntity> administrationConditions;
-    private Set<CodedEntity> completenesses;
```
Comment (esotrope) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/ReportOptions.java]: @dsnyder-fairway UI will hard-code these for now.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultAssessmentTypeService.java [hunk 2]
```diff
@@ -4,23 +4,44 @@
...
     private final AssessmentTypeRepository repository;
+    private final boolean interimAggregateReportsForStatewideUsersEnabled;
 
     @Autowired
-    DefaultAssessmentTypeService(final AssessmentTypeRepository repository) {
+    DefaultAssessmentTypeService(
+            final AssessmentTypeRepository repository,
+            @Value("${app.interim-aggregate-reports-for-statewide-users-enabled}") final boolean interimAggregateReportsForStatewideUsersEnabled) {
         this.repository = repository;
+        this.interimAggregateReportsForStatewideUsersEnabled = interimAggregateReportsForStatewideUsersEnabled;
```
Comment (esotrope) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultAssessmentTypeService.java]: default setting will be false (to disable interim aggregate reports for statewide users). This logic will have to be mirrored in the report generator service later.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/AssessmentTypeController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private final AssessmentTypeService service;
+
+    AssessmentTypeController(final AssessmentTypeService service) {
+        this.service = service;
+    }
+
+    @GetMapping
+    public Set<CodedEntity> get(@AuthenticationPrincipal final User user) {
+        return service.getAssessmentTypes(user);
+    }
```
Comment (wtritch) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/AssessmentTypeController.java]: Hmmm are there other options that will be user-specific?  Should we have a more general "/user/report-options" kind of endpoint to reduce chatter?

Comment (esotrope) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/AssessmentTypeController.java]: good point. what i can do is keep one endpoint and only cache a partial options object (the parts that arent user-specific)

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultUserReportOptionService.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        this.assessmentTypeService = assessmentTypeService;
+    }
+
+    @Override
+    public ReportOptions getReportOptions(final User user) {
+        return ReportOptions.builder()
+                .copy(reportOptionService.getReportOptions())
+                .assessmentTypes(assessmentTypeService.getAssessmentTypes(user))
+                .build();
+    }
```
Comment (esotrope) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultUserReportOptionService.java]: @wtritch added UserReportOptionService to add user-specific info to the report options payload.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/UserReportOptionController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import org.opentestsystem.rdw.olap.model.ReportOptions;
+import org.opentestsystem.rdw.olap.service.UserReportOptionService;
+import org.opentestsystem.rdw.reporting.common.web.security.User;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/userReportOptions")
```
Comment (esotrope) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/UserReportOptionController.java]: @dsnyder-fairway changed the endpoint

**Pull request merged**